### DISCUSSION TITLE
Why ALP compression algorithm need two numbers both exponent and factor?

### DISCUSSION THREAD

**Discussion by Lloyd-Pottiger** [duckdb/duckdb#11216] [Q&A]
I have watched this excellent sharing https://www.youtube.com/watch?v=GKG5b04o5Yc, and the code in this repo, thanks for your wonderful and creative job.

But I have a question for this, why ALP compression algorithm need two numbers both exponent and factor?

Since that, we will use one exponent and one factor for the whole vector, and the encoding and decoding work like

```shell
encode: n * 10^exponent * 10^(-factor) # which is equal to n * 10^(exponent-factor)
decode: n * 10^factor * 10^(-exponent) # which is equal to n * 10^(factor-exponent)
```

If I just store a value, for example, named as `mul` and equal to `exponent-factor`, the encoding and decoding can work like

```shell
encode: n * 10^mul
decode: n * 10^(-mul)
```

Can this also work? Or any other special consideration in your design?

**Comment by Tishj** [comment]
https://ir.cwi.nl/pub/33334/33334.pdf
You should ask the authors 👍

**Comment by Lloyd-Pottiger** [comment]
@Tishj Thanks!
